### PR TITLE
improve CLI tests

### PR DESCRIPTION
**Pull request by pmeier**
## What does this implement/fix?

Our current CLI tests are not sufficient, since our checking logic does not do what it is supposed to be doing. We only catch errors in the subprocess:

https://github.com/nebari-dev/nebari/blob/daecbcf37497b1b8b4e3f0f52667ababf3c023ad/tests/test_cli.py#L15-L23

However, `subprocess.call` does not fail on exit codes > 0:

```bash
$ ls foo
"foo": No such file or directory (os error 2)
$ echo $?
2
$ python -c 'import subprocess; subprocess.call(["ls", "foo"])'
ls: cannot access 'foo': No such file or directory
$ echo $?
0
```

This PR uses [`subprocess.run`](https://docs.python.org/3/library/subprocess.html#subprocess.run) instead, which is the [recommended way](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module) and has the `check=True` flag to enable error code checking.

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -32,28 +18,35 @@ def run_cli_cmd(command: str, working_dir: typing.Union[str, Path]):
 )
 def test_nebari_init(tmp_path, namespace, auth_provider, ci_provider, ssl_cert_email):
     """Test `nebari init` CLI command."""
-    command = f"nebari init local --project {PROJECT_NAME} --domain {DOMAIN_NAME} --disable-prompt"
+    command = [
```
Comment (pmeier) [inline tests/test_cli.py:21]: Minor refactoring. Instead of building a string here and splitting it again afterwards, let's just use a list from the start.

File: tests/test_cli.py [hunk 2]
```diff
@@ -67,7 +60,6 @@ def test_nebari_init(tmp_path, namespace, auth_provider, ci_provider, ssl_cert_e
         assert ci_cd.get("type", {}) == ci_provider
     else:
         assert ci_cd == ci_provider
-        ci_cd = config.get("ci_cd", None)
```
Comment (pmeier) [inline tests/test_cli.py:70]: Driveby: this wasn't used anywhere.

**Review by iameskild — APPROVED** [review]
Nice catch @pmeier! This looks good to me!
